src/xunit.analyzers/AssertEqualStringValueShouldNotUseGenericType.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Xunit.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AssertEqualStringValueShouldNotUseGenericType : AssertUsageAnalyzerBase
+    {
+        private static readonly HashSet<string> EqualMethods = new HashSet<string>(new[] { "Equal" });
+
+        public AssertEqualStringValueShouldNotUseGenericType() :
+            base(Constants.Descriptors.X2006_AssertEqualStringValueShouldNotUseGenericType, EqualMethods)
+        {
+        }
+
+        protected override void Analyze(SyntaxNodeAnalysisContext context, InvocationExpressionSyntax invocation, IMethodSymbol method)
+        {
+            var arguments = invocation.ArgumentList.Arguments;
+            if (arguments.Count != 2)
+                return;
+
+            var stringType = context.SemanticModel.Compilation.GetTypeByMetadataName(typeof(string).FullName);
```

Comment:
Use `compilation.GetSpecialType(SpecialType.System_String)` or maybe even the IsSpecialType method in the if statement instead of looking the type up at all.